### ISSUE TITLE
Split Parameter Discussion 

### ISSUE THREAD

**Issue by alexhallam** [alexhallam/Stemplot.jl#5]
> Add `split::Bool` parameter - break a stem across two or more lines if there's too many digits (or should this be `split::Int`, `maxdigits::Int`...?). For example:

```julia
julia> mystemplot
 0 | 0113
 1 | 0123456789
 2 |
 3 | 88
julia> stemplot!(mystemplot, split=true)
 0 | 0113
 1 | 012345
 1 | 6789
 2 |
 3 | 88
```

I don't think it is standard to see a single integer that gets two rows like is shown in the second example. I have seen truncation where a stem that has no leaves is cut out, but never a duplicated stem. That being said, if there is a reference/paper or examples in other software such as `R` that has this behavior I would not mind adding it. 

For clarity...

### Truncation

```julia
julia> stemplot(mystemplot)
 0 | 0113
 1 | 0123456789
 2 |
 3 | 88

julia> stemplot(mystemplot, trunc = true)
 0 | 0113
 1 | 0123456789
 3 | 88
```

**Comment by equinetic** [comment]
I had not heard of it myself – I saw it mentioned on a few ([1](http://spot.pcc.edu/~hmesa/math243/lecture/Stemplots.PDF), [2](http://learnalgebrafaster.com/split-stem-and-leaf-plot/)) pages I was perusing for ideas.

I think you're right in that this is not a common routine. The use case is obviously to limit output per stem, but one could typically adjust the scale to do that. If this remains on the docket (and perhaps it shouldn't) it'd probably be fairly low priority versus features like truncation.

**Comment by alexhallam** [comment]
I agree. Also, can you give an example of the `trim::Bool` parameter?

**Comment by equinetic** [comment]
The trim parameter would be to adjust the desired precision and possibly remove additional digits. Take the following output for example:
```julia
# Current
v = [111, 115, 121, 127, 135]
stemplot(v, scale=10)
``` 

> 11 | 15
> 12 | 17
> 13 | 5
> Key: 1|0 = 10
> The decimal is 1 digit(s) to the right of |

But maybe you're only interested in every 10. Setting `scale=100` gets you close but still shows you more information than you need. Now that I'm looking at this again I'm not sure if `trim` is the right name, and it'd need to be numeric, but this is the idea:

```julia
# With "trim" functionality
stemplot(v, scale=100, trim=10)
```

> 1 | 1020203040
> 
> Key: 1|0 = 100
> The decimal is 2 digit(s) to the right of |
> 

Maybe there'd be some other parameter or default logic to get rid of the 0's in that output so it's just "1 | 12234".

**Comment by equinetic** [comment]
Follow up on this, I think you're right. For how uncommon this operation would be it'd be a real pain to implement.
